token-contracts/WorkerToken/src/main/java/finance/omm/score/WorkerTokenImpl.java

```diff
@@ -0,0 +1,159 @@
+
+        BigInteger totalSupply = _initialSupply.multiply(pow(TEN, _decimals.intValue()));
+        Context.println(TAG + "| total_supply " + totalSupply);
+
+        this.totalSupply.set(totalSupply);
+        this.decimals.set(_decimals);
+        this.balances.set(Context.getCaller(), totalSupply);
+
+        Context.println(TAG + "| Started Worker Token:" + Context.getAddress());
+
+    }
+
+    public void onUpdate() {
+        Context.println(TAG + "| on update event:" + Context.getAddress());
+    }
+
+    @EventLog(indexed = 3)
+    public void Transfer(Address _from, Address _to, BigInteger _value,
+            byte[] _data) {
+    }
+
+    @External(readonly = true)
+    public String name() {
+        return "OMM " + TAG;
+    }
+
+    @External(readonly = true)
+    public String symbol() {
+        return WORKER_TOKEN_SYMBOL;
+    }
+
+    @External(readonly = true)
+    public BigInteger decimals() {
+        return this.decimals.get();
+    }
+
+    @External(readonly = true)
+    public BigInteger totalSupply() {
+        return this.totalSupply.get();
+    }
+
+    @External(readonly = true)
+    public BigInteger balanceOf(Address _owner) {
+        return this.balances.getOrDefault(_owner, ZERO);
+    }
+
+    @External
+    public void transfer(Address _to, BigInteger _value, @Optional byte[] _data) {
+        if (_data == null || _data.length == 0) {
+            _data = "None".getBytes();
+        }
+
+        if (!wallets.contains(_to)) {
+            this.wallets.add(_to);
+        }
+        this._transfer(Context.getCaller(), _to, _value, _data);
+    }
+
+    public void _transfer(Address _from, Address _to, BigInteger _value, byte[] _data) {
+
+        // Checks the sending value and balance.
+        if (_value.compareTo(ZERO) <= 0) {
+            Context.revert(TAG + ": Transferring value should be greater than zero");
+        }
+
+        BigInteger balanceFrom = this.balances.getOrDefault(_from, ZERO);
+        Context.println(TAG + "| transferring: " + _value + " - from sender balance: " + balanceFrom);
+
+        if (balanceFrom.compareTo(_value) < 0) {
+            Context.revert(TAG + " : Out of balance: " + _value);
+        }
+
+        this.balances.set(_from, balanceFrom.subtract(_value));
+        this.balances.set(_to, this.balances.getOrDefault(_to, ZERO).add(_value));
+
+        if (this.balanceOf(_from).equals(ZERO)) {
+            this.wallets.remove(_from);
+        }
+
+        if (_to.isContract()) {
+            // If the recipient is SCORE,
+            // then calls `tokenFallback` to hand over control.
+            Context.call(_to, "tokenFallback", _from, _value, _data);
+        }
+        // Emits an event log `Transfer`
+        this.Transfer(_from, _to, _value, _data);
+    }
+
+    @External(readonly = true)
+    public List<Address> getWallets() {
+        return arrayDbToList(this.wallets.getEntries());
+    }
+
+    private static BigInteger pow(BigInteger base, int exponent) {
```

Comment:
I think we has already pow method in MathUtils.

Reply:
done